xarray/core/rolling.py

```diff
@@ -151,34 +151,21 @@ def __init__(self, obj, min_periods=None, center=False, **windows):
         """
         super(DataArrayRolling, self).__init__(obj, min_periods=min_periods,
                                                center=center, **windows)
-        self.window_indices = None
-        self.window_labels = None
 
-        self._setup_windows()
+        self.window_labels = self.obj[self.dim]
 
     def __iter__(self):
-        for (label, indices) in zip(self.window_labels, self.window_indices):
-            window = self.obj.isel(**{self.dim: indices})
+        _stops = np.arange(1, len(self.window_labels) + 1)
```

Comment:
nit: no need to preface with an underscore for variables that are already limited in scope to this method.